src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/ReferenceBlockConcordance.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    private void evaluateEndOfContig() {
+        if (currentTruthVariantContext != null && currentEvalVariantContext != null) {
+            int blockStart = Math.max(currentTruthVariantContext.getStart(), currentEvalVariantContext.getStart());
+            int blockEnd = Math.min(currentTruthVariantContext.getEnd(), currentEvalVariantContext.getEnd());
+            int jointBlockLength = blockEnd - blockStart + 1;
+            if (jointBlockLength > 0) {
+                confidenceConcordanceHistogram.increment(new Pair<>(currentTruthVariantContext.getGenotype(0).getGQ(), currentEvalVariantContext.getGenotype(0).getGQ()).toString(), jointBlockLength);
+            }
+        }
+
+        currentTruthVariantContext = null;
+        currentEvalVariantContext = null;
+        currentContig = null;
+    }
+
+    private void evaluateNewContig(TruthVersusEval truthVersusEval) {
+        // If not beginning of file
+        if (currentContig != null) {
+            evaluateEndOfContig();
+        }
+
+        currentContig = truthVersusEval.getTruthIfPresentElseEval().getContig();
+    }
+
+    @Override
+    protected void apply(TruthVersusEval truthVersusEval, ReadsContext readsContext, ReferenceContext refContext) {
+        // New contig or beginning of file
+        if (!truthVersusEval.getTruthIfPresentElseEval().getContig().equals(currentContig)) {
+            evaluateNewContig(truthVersusEval);
+        }
+
+        // Evaluate only when currently seeing two NON_REF blocks
+        if (currentTruthVariantContext != null && currentEvalVariantContext != null) {
+            int blockStart = Math.max(currentTruthVariantContext.getStart(), currentEvalVariantContext.getStart());
+            int blockEnd = Math.min(currentTruthVariantContext.getEnd(), currentEvalVariantContext.getEnd());
+            int jointBlockLength = blockEnd - blockStart + 1;
+            // It is possible that jointBlockLength is negative if there is a gap in one file and the start of a new block in the other file.
+            // Since there is no overlap though, we can just skip that case.
+            if (jointBlockLength > 0) {
+                confidenceConcordanceHistogram.increment(new Pair<>(currentTruthVariantContext.getGenotype(0).getGQ(), currentEvalVariantContext.getGenotype(0).getGQ()).toString(), blockEnd - blockStart + 1);
+            }
+
+            int currentPosition = truthVersusEval.getTruthIfPresentElseEval().getStart();
+            if (truthVersusEval.hasTruth() || currentPosition >= currentTruthVariantContext.getEnd()) {
+                currentTruthVariantContext = null;
+            }
+            if (truthVersusEval.hasEval() || currentPosition >= currentEvalVariantContext.getEnd()) {
+                currentEvalVariantContext = null;
+            }
+        }
+
+        // Truth
+        if (truthVersusEval.hasTruth() && isNonRef(truthVersusEval.getTruth())) {
+            currentTruthVariantContext = truthVersusEval.getTruth();
```

Comment:
could the setting of `currentTruthVariantContext` and `currentEvalVariantContext` be moved before the incrementing of `confidenceConcordanceHistogram`.  Then you wouldn't need to additionally increment `confidenceConcordanceHistogram` in `evaluateEndOfContig`, or call `evaluateEndOfContig` from `onTraversalSuccess`